apps/src/applab/designMode.js

```diff
@@ -154,11 +154,12 @@ designMode.resetElementTray = function (allowEditing) {
  * Given an input value produce a valid css value that is
  * either in pixels or empty.
  */
-function appendPx(input) {
+export function appendPx(input) {
   // Don't append if we already have a px
   if (/px/.test(input)) {
     return input;
   }
+  input = input.toString();
   return input ? input + 'px' : '';
```

Comment:
Your solution behaves correctly for good input but I'm not sure it expresses the spirit of the check as well as it could.

The proper domain of the function here is the union of three sets: Numbers (`42`), strings containing numbers (`"42"`), and strings containing numbers that already have `"px"` appended (`"42px"`).  You've covered these cases in your tests. :+1: 

Reading the line
```javascript
return input  ? input + 'px' : '';
```
I think the original author's intent is to catch any input _outside_ the function's proper domain (like `null` or `false`) and make sure to return empty string.  Your change fixes the problem where `0`, which is in the domain, was treated like it out.  But the original author _also_ missed a lot of cases where values outside the domain are treated like they are in.

```js
appendPx('foo') // "foopx"
appendPx({foo: 'bar'}) // "[object Object]px"
appendPx([1, 2, 3]) // "1,2,3px"
```

It might seem like overkill to care about this for a little utility function, but the more uncertainty we can remove from our code the less often we write bugs.  I'd add test cases for a few of these "bad input" cases in addition to the ones you've got, to make sure this function has the expected behavior in odd circumstances.

Then, I'd guide your implementation by thinking about what we actually want to check here.  If the goal is to check whether we've actually got a number or a string containing a number, functions like [`parseInt`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt), [`parseFloat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseFloat) and [`isNaN`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN) are going to help a lot.  I think I'd implement it this way:

```javascript
const parsed = parseFloat(input);
return (isNaN(parsed) || isArray(input)) ? '' : (parsed + 'px');
```